### PR TITLE
Use Sequelize.STRING with length

### PR DESCRIPTION
**Pull request by catto**
MySQL requires that an indexed column including primary key is fixed length.
Sequelize will generate a DDL with no length option if we specify `Sequelize.TEXT` as data type.
Screwdriver mainly uses string hash as ID, so the current version  doesn't work with MySQL. (screwdriver-cd/screwdriver#325)

This PR fixes this issue by using `Sequelize.STRING(len)` for fixed-length string column.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -80,11 +80,24 @@ function encodeToDialect(dialect, content, model) {
...
  * @param  {String}          type    Joi type
+ * @param  {Array}           rules   Joi rules array
  * @return {SequelizeType}           Type to use in Sequelize
  */
-function getSequelizeTypeFromJoi(dialect, type) {
+function getSequelizeTypeFromJoi(dialect, type, rules) {
+    let length;
+
+    // Get the column length if length/max rule is included in dataschema
+    if (rules) {
```
Comment (d2lam) [inline index.js]: Isn't rules always passed in?

**Comment by d2lam** [comment]
The rest LGTM 👍

File: index.js (hunk 1 above)
Comment (catto) [inline index.js]: Yes. But `joi` doesn't create the `rules` key when schema has no rule such as length.
```
> require('joi').string().describe()
{ type: 'string', invalids: [ '' ] }
```
I'll change L167 not to pass undefined to `getSequelizeTypeFromJoi`.

**Comment by d2lam** [comment]
Hmmm No SD build created for this as well because of: https://github.com/screwdriver-cd/screwdriver/issues/335

**Pull request merged**